### PR TITLE
Rework enums to be PyPy-compatible

### PR DESCRIPTION
**Pull request by jlaine**

### CODE REVIEW

**Comment by jlaine** [comment]
OK, well this broke some things on Python 3.4 as dicts are not ordered.

**Comment by jlaine** [comment]
PyPy 2.7 fully passes the test suite! PyPy 3.5 and 3.6 have two failures, because Travis's PyPy versions don't seem to work correctly with numpy, the multiarray module fails to import.

File: tests/test_enums.py [hunk 1]
```diff
@@ -21,8 +21,6 @@ def test_basics(self):
 
         cls = self.define_foobar()
 
-        self.assertIsInstance(cls, EnumType)
```
Comment (mikeboers) [inline tests/test_enums.py]: Does this not still work?

**Comment by mikeboers** [comment]
I pushed a couple small changes to your branch. Feel free to disregard them if you disagree.

I just thought it would be keen to maintain the normal `type.__new__` API.

Did you happen to do a version at some point where EnumType was still cdef? I feel like that would still work. Although, I could/should not care because that wouldn't affect the use of the enums in a significant way.

File: tests/test_enums.py (hunk 1 above)
Comment (mikeboers) [reply tests/test_enums.py]: It does in my version. The commits pushed below have this. Sorry if this breaks CI. 😉

**Pull request merged**